vowpalwabbit/object_pool.h

```diff
@@ -114,13 +114,14 @@ struct no_lock_object_pool
     }
   }
 
-  std::queue<T*> m_pool;
-  std::vector<std::pair<T*, T*>> m_chunk_bounds;
-  std::vector<std::unique_ptr<T[]>> m_chunks;
   TInitializer m_initializer;
   TCleanup m_cleanup;
-  size_t m_initial_chunk_size = 0;
-  size_t m_chunk_size = 8;
+  const size_t m_initial_chunk_size = 0;
```

Comment:
https://www.reddit.com/r/cpp/comments/8wbeom/coding_guideline_avoid_const_member_variables/

We should avoid const members